src/System.Net.Security/ref/System.Net.Security.cs

```diff
@@ -167,6 +167,8 @@ public partial class SslStream : System.Net.Security.AuthenticatedStream
         public override int ReadTimeout { get { throw null; } set { } }
         public virtual System.Security.Cryptography.X509Certificates.X509Certificate RemoteCertificate { get { throw null; } }
         public virtual System.Security.Authentication.SslProtocols SslProtocol { get { throw null; } }
+        [CLSCompliantAttribute(false)]
+        public virtual System.Net.Security.TlsCipherSuite? NegotiatedCipherSuite { get { throw null; } }
```

Comment:
Remind me why this is nullable?  Why is this nullable but something like SslProtocol isn't?

Reply:
I wondered about this also.  But I watched the YouTube video recording of the API review.  They discussed this extensively.

It is nullable in case a platform (Xamarin etc.) is unable to ever return a proper value for this.

Reply:
I'll try to find time to go watch it again.  I'm not seeing how it's different from the other properties like KeyExchangeStrength, CipherAlgorithm, etc.  What's special about TlsCipherSuite that a platform couldn't return a valid value for this but could for all of those?  And is this speculative concern, or we know about cases where it can't actually be done?

Reply:
@bartonjs Can you comment on this?

Reply:
The difference is that the zero ciphersuite is a legal protocol answer; so there really should be a state of indicating that the answer is "not known", vs "known to be TLS_NULL_NULL_WITH_NULL".

The existing properties probably should have been nullable, too, but they were designed when Win32 was the only supported platform.  This property (at the protocol level) drives all the rest; so this one being null would (except in a weird platform where we get the breakdown products but not the actual answer) indicate that zero on those values meant "unknown" vs "known to be zero".

Reply:
> so there really should be a state of indicating that the answer is "not known"

When will that happen? Is there a platform we know about where it can't be known?

Reply:
From watching the API review, there isn't one identified yet.  But the problem is that it would be a breaking change for the API to switch to a nullable type.

But IMHO, making this nullable makes the API less usable (you have to check .HasValue).  I think it's not a terrible thing to just let the value be TLS_NULL_NULL_WITH_NULL even if it is unknown.  That's a simple thing to document on platforms where it doesn't really work.

Reply:
I don't know of any where it wouldn't work, but I don't know the landscape beyond Windows, macOS, and OpenSSL.  If the various Xamarin things that exist either all can get the data or just utterly don't support SslStream, then it's probably reasonable to assume the information can always be gotten and the nullability won't matter and can be dropped from the API.

Reply:
reverted nullable from API